lib/HLSL/DxilReduceMSAAToSingleSample.cpp

```diff
@@ -59,11 +59,13 @@ bool DxilReduceMSAAToSingleSample::runOnModule(Module &M)
       if (isa<ConstantInt>(createHandle.get_rangeId())){
         unsigned rangeId = cast<ConstantInt>(createHandle.get_rangeId())->getLimitedValue();
         auto Resource = DM.GetSRV(rangeId);
-        if (Resource.GetKind() == DXIL::ResourceKind::Texture2DMS || Resource.GetKind() == DXIL::ResourceKind::Texture2DMSArray) {
-          //todo: no apparent constant definition for the mip-level/sample-index operand index (the "2"):
-          // https://github.com/Microsoft/DirectXShaderCompiler/blob/master/docs/DXIL.rst#textureload
-          instruction->setOperand(2, HlslOP->GetI32Const(0));
-          Modified = true;
+        if (static_cast<DXIL::ResourceClass>(createHandle.get_resourceClass_val()) == DXIL::ResourceClass::SRV) {
```

Comment:
Need to move `auto Resource = DM.GetSRV(rangeId);` into this if body, otherwise, that will be potentially out of range and invalid (could assert?).